paper_framework_sources/p1897_initial_executor_algorithms_for_cpp23.md

```diff
@@ -694,21 +748,123 @@ float r = sync_wait(
+   and that does not include a declaration of `execution::share`.
+
+ * Otherwise constructs a receiver, `r` and passes that receiver to `execution::connect(s, r)` resulting in an `operation_state` `os`.
+   Constructs some shared state, `shr` to store the completion result(s) of `s`.
+
+   * If `set_value(r, ts)` is called stores `ts` in `shr`.
+   * If `set_error(r, e)` is called, stores `e` in `shr`.
+   * If `set_done(r)` is called stores the done result in `shr`.
+
+   When some `output_receiver` has been passed to `connect` on the returned `sender`, resulting in an `operation_state` `os2` and one of the above has been called on `r`:
+   * If `r` was satisfied with a call to `set_value`, call `set_value(output_receiver, ts...)`
+   * If `r` was satisfied with a call to `set_error`, call `set_error(output_receiver, e)`.
+   * If `r` was satisfied with a call to `set_done`, call `execution::set_done(output_receiver)`.
+
+ * When `start` is called on `os2`, call `execution::start(os)`.
+
+## execution::let
+
+### Overview
+`let` is a sender adapter that takes a `sender` and an invocable and returns a `sender` that keeps the completion result of the incoming sender alive for the duration of the algorithm returned by the invocable and makes that value available to the invocable.
+
+Signature:
+```cpp
+template <execution::sender S, std::invocable F>
+see-below let(S s, F f);
+```
+
+where `S<T...>` and `S<T2>` are implementation-defined types that is represent senders that send a value of type list `T...` or `T2` respectively in their value channels.
+Note that in the general case there may be many types `T...` for a given `sender`, in which case the invocable may have to represent an overload set.
+
+*[ Example:*
+```cpp
+int r = sync_wait(
+  just(3) |
+  let([](int& let_v){
+    return just(4) | transform([&](int v){return let_v + v;})));
+// r==7
+```
+
+### Wording
+The name `execution::let` denotes a customization point object.
+For some subexpressions `s` and `f`, let `S` be a type such that `decltype((s))` is `S` and `decltype((f))` is `F`.
+The expression `execution::let(s, f)` is expression-equivalent to:
+
+ * s.let(f), if that expression is valid, if `s` satisfies `sender` and `f` satisfies `invocable`.
+ * Otherwise, `let(s, f)`, if that expression is valid,, if `s` satisfies `sender` and `f` satisfies `invocable` with overload resolution performed in a context that includes the declaration
+```
+    void let() = delete;
+```
+  and that does not include a declaration of `execution::let`.
+ * Otherwise constructs a receiver, `r` and passes that receiver to `execution::connect(S, r)` returning an `operation_state` `os` such that
+   When some `output_receiver` has been passed to `connect` on the returned `sender` returning some `operation_state` `os2`:
+
+   * If `set_value(r, e)` is called, calls `std::invoke(f, ts...)` to return some `invoke_result`, and calls `execution::start(execution::connect(invoke_result, output_receiver))`.
```

Comment:
We want to be explicit about `ts...` being copied into the operation-state here and then lvalues to these copies being passed to `f()` rather than `ts...` that were passed to `set_value()` originally.

Also need to be explicit about the lifetime of these copies. ie. that their destructors are sequenced after the operation produced by `f()` completes.